### PR TITLE
MF-518: action menu (work in progress)

### PR DESCRIPTION
**Pull request by seankwon**
# Description

Still a work in progress but opening a draft to show where i am. needs further requirements

## the menu itself
Screen Shot 2021-05-12 at 9 16 38 PM

# menu with active item
Screen Shot 2021-05-12 at 9 16 41 PM

# hiding menu on mobile and showing the pen (but what does clicking the pen do?)
Screen Shot 2021-05-12 at 9 16 50 PM

### CODE REVIEW

File: packages/esm-patient-chart-app/src/ui-components/action-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import React, { useCallback } from "react";
+import {
+  Edit20,
```
Comment (FlorianRappl) [inline packages/esm-patient-chart-app/src/ui-components/action-menu.tsx:3]: Can you please use direct submodule imports? Taking the whole icons-react lib will result in a potential unnecessary increase in bundle size.

File: packages/esm-patient-chart-app/src/ui-components/action-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  Notification20,
+  ShoppingBag20,
+  Document20,
+} from "@carbon/icons-react";
+import Button from "carbon-components-react/es/components/Button";
+import { HeaderPanel } from "carbon-components-react";
```
Comment (FlorianRappl) [inline packages/esm-patient-chart-app/src/ui-components/action-menu.tsx:9]: Please use direct submodule import.

File: packages/esm-patient-chart-app/src/ui-components/action-menu.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    config.state.type,
+  ]);
+  const onDocumentsClick = useCallback(
+    createToggle(ActionMenuTypes.DOCUMENTS),
+    [config.state.type]
+  );
+
+  const menu = isDesktop(layout) ? (
+    <aside className={styles.rightSideNav}>
+      <Button
```
Comment (FlorianRappl) [inline packages/esm-patient-chart-app/src/ui-components/action-menu.tsx:43]: Should these buttons be hard coded in here?

Some of them don't seem to be related to patient chart at all. Please offer a proper extension slot.

**Review by FlorianRappl — COMMENTED** [review]
Overall this seems to be on a good way.

File: packages/esm-patient-chart-app/src/ui-components/action-menu.tsx (hunk 1 above)
Comment (seankwon) [reply packages/esm-patient-chart-app/src/ui-components/action-menu.tsx:3]: i actually had no idea how to do that for carbon icons... can you show me an example?

File: packages/esm-patient-chart-app/src/ui-components/action-menu.tsx (hunk 3 above)
Comment (seankwon) [reply packages/esm-patient-chart-app/src/ui-components/action-menu.tsx:43]: hmmm gotcha - time to finally learn extensions... will get that to you soon. gonna have to think through creating an extension slot for the drawer content as well, i think

File: packages/esm-patient-chart-app/src/ui-components/action-menu.tsx (hunk 1 above)
Comment (FlorianRappl) [reply packages/esm-patient-chart-app/src/ui-components/action-menu.tsx:3]: Pattern:

```js
import <AppropriateNameForTheDefaultExport> from '@carbon/icons-react/es/<name of icon>/<size of icon>';
```

Example:

```js
import ChevronDown16 from '@carbon/icons-react/es/chevron--down/16
```
